pkg/provisioners/channel_util.go

```diff
@@ -170,26 +170,42 @@ func CreateVirtualService(ctx context.Context, client runtimeClient.Client, chan
 }
 
 func UpdateChannel(ctx context.Context, client runtimeClient.Client, u *eventingv1alpha1.Channel) error {
+	objectKey := runtimeClient.ObjectKey{Namespace: u.Namespace, Name: u.Name}
 	channel := &eventingv1alpha1.Channel{}
-	err := client.Get(ctx, runtimeClient.ObjectKey{Namespace: u.Namespace, Name: u.Name}, channel)
-	if err != nil {
+
+	if err := client.Get(ctx, objectKey, channel); err != nil {
 		return err
 	}
 
-	updated := false
+	channelChanged := false
+
 	if !equality.Semantic.DeepEqual(channel.Finalizers, u.Finalizers) {
 		channel.SetFinalizers(u.ObjectMeta.Finalizers)
-		updated = true
+		if err := client.Update(ctx, channel); err != nil {
+			return err
+		}
+
+		channelChanged = true
 	}
 
-	if !equality.Semantic.DeepEqual(channel.Status, u.Status) {
-		channel.Status = u.Status
-		updated = true
+	if equality.Semantic.DeepEqual(channel.Status, u.Status) {
+		return nil
 	}
 
-	if updated {
-		return client.Update(ctx, channel)
+	if channelChanged {
+		// Refetch
```

Comment:
Why are we refetching? 

What does the call `client.Status().Update(ctx, channel)` use inside `channel`? Is it just `status` and apiVersion, groupKind, name, and namespace?

Reply:
I believe the entire object is posted to the status endpoint - even though 'spec' is ignored I'm assuming that an object's `resourceVersion` is used by the API server.

If we don't refetch after an update I'm assuming the `resourceVersion` won't match the server and thus the subsequent status update will fail since we're applying the update.